### PR TITLE
importccl: restart IMPORT on worker node failure

### LINKED ISSUE #25480: import: allow worker node failure
During IMPORT the job coordinator can go down and the job will be restarted. However if another node goes down I think the job will be marked as failed. In this case we should be aware that this is a transient error and restart the job so it makes a new plan with healthy nodes.

### LINKED ISSUE #25866: jobs: retry transient errors
A context canceled error in a job should restart it instead of not. It is possible this will get into an infinite loop if the context canceled error is masking an actual error, but that's not what I've observed so far.

### PR DESCRIPTION
**Pull request by madelynnblue**
Attempt to detect a context canceled error in IMPORT which is caused by
a node going away in the dist SQL run. Send a special error back to the
job registry indicating a restart should happen instead of a failure.

We are shipping this with a skipped test because it is flakey. We are
ok doing that because it is still better than what we had before in many
cases, just not all. We will work to improve the other things so that we
can correctly detect when IMPORT can be restarted due to a node outage,
which will allow us to unskip this test.

Fixes #25866
Fixes #25480

Release note (bug fix): IMPORT now detects node failure and will restart
instead of fail.

### CODE REVIEW

**Comment by mjibson** [comment]
PTAL @dt

File: pkg/ccl/importccl/import_stmt_test.go [hunk 1]
```diff
@@ -1159,6 +1159,80 @@ func TestImportControlJob(t *testing.T) {
...
 
+// TestImportWorkerFailure tests that IMPORT can restart after the failure
+// of a worker node.
+func TestImportWorkerFailure(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	// TODO(mjibson): Although this test passes most of the time it still
+	// sometimes fails because not all kinds of failures caused by shutting a
+	// node down are detected and retried.
+	t.Skip()
```
Comment (dt) [inline pkg/ccl/importccl/import_stmt_test.go]: nit: i generally like to pass a brief explanation (or filed issue #) to Skip.

**Review by dt — APPROVED** [review]

File: pkg/ccl/importccl/import_stmt_test.go (hunk 1 above)
Comment (mjibson) [reply pkg/ccl/importccl/import_stmt_test.go]: fixed

**Pull request merged**